### PR TITLE
Refactor evms.contracts

### PR DESCRIPTION
**Pull request by Hosuke**
Make [`blockchain`, `address`] unique in final `evms.contracts`.

### CODE REVIEW

**Comment by Hosuke** [comment]
[evms.contracts preview](https://dune.com/queries/4150578)

File: dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql [hunk 1]
```diff
@@ -32,25 +32,29 @@
...
+            , "from"
+            , code
+            , name
+            , namespace
+            , dynamic
+            , base
+            , factory
+            , detection_source
+            , created_at
+            , row_number() over (partition by address order by created_at desc) as duplicates_rank
```
Comment (jeff-dude) [inline dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql:52]:
```suggestion
            , row_number() over (partition by blockchain, address order by created_at desc) as duplicates_rank
```

Comment (jeff-dude) [reply dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql:52]: if we're unique on these two fields, don't they both need added to partition in row number?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql (hunk 1 above)
Comment (Hosuke) [reply dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql]: You are right, I did the partition by address before union by chain.
But would be safer to partition by blockchain and address together.

File: dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql [hunk 2]
```diff
@@ -32,25 +32,29 @@
...
+            , "from"
+            , code
+            , name
+            , namespace
+            , dynamic
+            , base
+            , factory
+            , detection_source
+            , created_at
+            , row_number() over (partition by blockchain, address order by created_at desc) as duplicates_rank
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql:52]: Sorry, at the partition moment, the `blockchain` feild is not existed yet.

**Comment by jeff-dude** [comment]
understood on address being unique per chain level contracts table, thank you for clarification.

however, now that we do this approach, we should remove `created_at` from unique key in model. then we should be all good to go.

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
will run this in prod in a few minutes, should refresh in next 20-30 minutes hopefully

**Pull request merged**